### PR TITLE
249 add maximum number of ballons

### PR DESCRIPTION
**Pull request by rodrigoAMF**
As requested on https://github.com/TheAlgorithms/C/issues/249

Add solution to problem 1189 Maximum Number of Ballons available at: https://leetcode.com/problems/maximum-number-of-balloons/

Also updated README from leetcode folder.

### CODE REVIEW

File: leetcode/src/1189.c [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    /* Divide by 2 the repeted letters */
+    count_letters[2] /= 2;
+    count_letters[3] /= 2;
+    count_letters[4] /= 2;
+    count_letters[5] /= 2;
+    
+    /* Max number of times which we can write ballon is equal to min value of letters on count_letter */
+    int min_counter_ballons;
+    for(int i = 0; i < 7; i++){
+        if(i == 0){
```
Comment (danghai) [inline leetcode/src/1189.c]: I think we do not need the `if-condition` here. We just initialize the variable `min_counter_ballons = count_letters[0]` outside the loop

File: leetcode/src/1189.c [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            count_letters[3]++;
+        }else if(text[i] == 'o') {
+            count_letters[4]++;
+            count_letters[5]++;
+        }else if(text[i] == 'n') {
+            count_letters[6]++;
+        }
+    }
+    
+    /* Divide by 2 the repeted letters */
```
Comment (danghai) [inline leetcode/src/1189.c]: We do not need to count 2 times for repeated character.

File: leetcode/src/1189.c [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        3 -> l,
+        4 -> o,
+        5 -> o,
+        6 -> n
+    */
+    int count_letters[7] = {0};
+    
+    /* Count the number of each letter */
+    for(int i = 0; text[i] != '\0'; i++){
+        if(text[i] == 'b') {
```
Comment (danghai) [inline leetcode/src/1189.c]: I think it is better to use *ptr for traversing the string

**Comment by danghai** [comment]
Hi @rodrigoAMF . It is a good solution with explanation. I have some enhancement.
```
int maxNumberOfBalloons(char * text){
    /* 
        0 -> b, 
        1 -> a,
        2 -> l,
        3 -> o,
        4 -> n
    */
    int count_letters[5] = {0};
    int i, min_counter_ballons;
    
    for (char *ptr = text; *ptr; ptr++) {
        if(*ptr == 'b') {
            count_letters[0]++;
        }else if(*ptr == 'a') {
            count_letters[1]++;
        }else if(*ptr == 'l') {
            count_letters[2]++;
        }else if(*ptr == 'o') {
            count_letters[3]++;
        }else if(*ptr == 'n') {
            count_letters[4]++;
        }
    }
    
    /* Divide by 2 the repeted letters */
    count_letters[2] /= 2;
    count_letters[3] /= 2;
    
    /* Max number of times which we can write ballon is equal to min value of letters on count_letter */
    min_counter_ballons = count_letters[0];
    for(i = 1; i < 5; i++){
        if(count_letters[i] < min_counter_ballons)
            min_counter_ballons = count_letters[i];
    }
    
    return min_counter_ballons;
}
```

File: leetcode/src/1189.c (hunk 1 above)
Comment (rodrigoAMF) [reply leetcode/src/1189.c]: Yes good point thanks!

File: leetcode/src/1189.c (hunk 3 above)
Comment (rodrigoAMF) [reply leetcode/src/1189.c]: With that modification the code seems more readable but just for curiosity does this make any difference in terms of optimization?

File: leetcode/src/1189.c (hunk 2 above)
Comment (rodrigoAMF) [reply leetcode/src/1189.c]: That's true thanks for pointing out!

**Comment by rodrigoAMF** [comment]
Hi @danghai thanks for your time reviewing my code! I've updated the code with your enhancements.

File: leetcode/src/1189.c (hunk 3 above)
Comment (danghai) [reply leetcode/src/1189.c]: Ah it does not have any optimization. It is just more readable, and safe.

**Comment by danghai** [comment]
Thank you. I hope you can contribute more leetcode problem.

**Pull request merged**